File: 프로그래머스/Lv.0/다항식 더하기/120863.java
class Solution {
    public String solution(String polynomial) {
        String answer = "";
        
        String[] arr = polynomial.split(" ");
        int num1 = 0, num2 = 0;
        
        for (int i=0; i<arr.length; i++) {
            if ("+".equals(arr[i])) continue;
            
            if (arr[i].contains("x")) {
                num1 += (arr[i].equals("x")) ? 1 : Integer.parseInt(arr[i].split("x")[0]);
            } else {
                num2 += Integer.parseInt(arr[i]);
            }
        }
        
        answer += num1 > 0 ? (num1 > 1 ? num1 : "") + "x" : "";
        answer += num2 > 0 ? (num1 > 0 ? " + " : "") + num2 : "";
        
        return answer;
    }
}

File: 프로그래머스/Lv.0/다항식 더하기/120863.js
function solution(polynomial) {
    let answer = '';
    
    let num1 = 0, num2 = 0;
    polynomial.split(" + ").forEach(v => {
       if (v.indexOf("x") > -1) {
           num1 += (v === "x") ? 1 : Number(v.split("x")[0]);
       } else {
           num2 += Number(v);
       }
    });
    
    answer += num1 > 0 ? (num1 > 1 ? num1 : "") + "x" : "";
    answer += num2 > 0 ? (num1 > 0 ? " + " : "") + num2 : "";
    
    return answer;
}
